File: VB/XtraReport1.vb
Imports System
Imports DevExpress.XtraReports.UI

' ...
Namespace DetailReportsSum

    Public Partial Class XtraReport1
        Inherits XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private GrandTotals As Decimal = 0

        Private Sub xrlFreightTotal_SummaryCalculated(ByVal sender As Object, ByVal e As TextFormatEventArgs)
            If e.Value IsNot Nothing Then GrandTotals += Convert.ToDecimal(e.Value)
        End Sub

        Private Sub xrlFreightGrandTotal_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs)
            xrlFreightGrandTotal.Text = String.Format("{0:c2}", GrandTotals)
        End Sub
    End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using DevExpress.XtraReports.UI;
// ...

namespace DetailReportsSum {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        decimal GrandTotals = 0;

        private void xrlFreightTotal_SummaryCalculated(object sender, TextFormatEventArgs e) {
            if(e.Value != null)
                GrandTotals += Convert.ToDecimal(e.Value);
        }

        private void xrlFreightGrandTotal_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            xrlFreightGrandTotal.Text = string.Format("{0:c2}", GrandTotals);
        }

    }
}
